docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+# 1. Create a Basic Text Classifier Neural Network with TensorFlow
+
+In this guide, we use the [Alteryx Python SDK](https://pypi.org/project/ayx-python-sdk/), [Alteryx UI-SDK](https://github.com/alteryx/dev-harness), and [Alteryx Plugin CLI](https://pypi.org/project/ayx-plugin-cli/) to illustrate how to create, train, and call a Tensorflow Keras Neural Network model in a workflow.
+To achieve this, We'll reference Tensorflow's [official tutorial](https://www.tensorflow.org/tutorials/keras/text_classification) and recreate the Basic Text Classifier as a workflow tool!
+
+Upon completing this guide, you will know how to:
+- Develop using the Python and UI SDKs
+- Exchange data between UI and Python SDK processes; during both `UPDATE` and `RUN` modes
+- Use metaprogramming, Python SDK, and the UI SDK to create an editable configuration and dynamic behavior between or during workflow runs
+- Package and use 3rd party libraries from `npm` and `pypi` in a tool
+- Develop "ready to use" Keras models from scratch, including models for deploying into production workflows.
+
+- [1. Create a Basic Text Classifier Neural Network with Tensorflow](#1-create-a-basic-text-classifier-nueral-network-with-tensorflow)
+  - [1.1. Prerequisites](#11-prerequisites)
+  - [1.2. Overview](#12-overview)
+    - [1.2.1. Data](#121-data)
+    - [1.2.2. Preview](#122-preview)
+    - [1.2.3. TRAIN](#123-train)
+    - [1.2.4. Predict](#124-predict)
+    - [1.2.5. Create a Plugin](#125-create-a-plugin)
+      - [1.2.5.1. UI/TextClassifier/src/index.tsx](#1251-uitextclassifiersrcindextsx)
+      - [1.2.5.2. backend/ayx\_plugins/text\_classifier.py](#1252-backendayx_pluginstext_classifierpy)
+  - [1.3. Backend: A Naive Tensorflow Text Classifier](#13-backend-a-naive-tensorflow-text-classifier)
+    - [1.3.1. Data Mode](#131-data-mode)
+      - [Supplemental: Error Handling in the SDK](#supplemental-error-handling-in-the-sdk)
+      - [Adding a custom method: `setup_data()`](#adding-a-custom-method-setup_data)
+    - [](#)
+
+
+## 1.1. Prerequisites
+
+While we do our best to provide enough information here for anyone with general scripting experience to follow along, the recommended prereqs are provided below for general debugging and troubleshooting knowledge outside of the scope possible here:
```

Comment:
TW review:

This guide aims to provide enough information for anyone with general coding experience to follow along. We recommend these prerequisites for general debugging and troubleshooting outside of the scope of this guide:

Reply:
Feedback has been good overall. Just to call out tho, I struggled with a good wording here for awhile and this is much better haha. Thanks!